app/src/main/java/com/epfl/beatlink/ui/profile/EditProfile.kt

```diff
@@ -0,0 +1,181 @@
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.epfl.beatlink.R
+import com.epfl.beatlink.ui.authentication.CustomInputField
+import com.epfl.beatlink.ui.components.CornerIcons
+import com.epfl.beatlink.ui.navigation.BottomNavigationMenu
+import com.epfl.beatlink.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.epfl.beatlink.ui.navigation.NavigationActions
+import com.epfl.beatlink.ui.theme.LightGray
+import com.epfl.beatlink.ui.theme.PrimaryGradientBrush
+import com.epfl.beatlink.ui.theme.lightThemeBackground
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun EditProfileScreen(navigationActions: NavigationActions) {
+  var name by remember { mutableStateOf("This is the current name") }
+  var description by remember {
+    mutableStateOf(
+        "This is the current description and it is very long because didier is such a cool guy that he has a lot of things to say about himself")
+  }
+  val scrollState = rememberScrollState()
+  val maxDescriptionLength = 100
+  val maxUsernameLength = 20
+  Scaffold(
+      modifier = Modifier.testTag("editProfileScreen"),
+      topBar = {
+        TopAppBar(
+            title = {
+              Text(
+                  text = "Edit profile",
+                  textAlign = TextAlign.Center,
+                  color = MaterialTheme.colorScheme.primary,
+                  style = MaterialTheme.typography.headlineLarge,
+                  fontSize = 20.sp,
+                  fontWeight = FontWeight.Bold,
+                  modifier =
+                      Modifier.fillMaxWidth().offset(x = (-28).dp).testTag("editProfileTitle"))
+            },
+            navigationIcon = {
+              CornerIcons(
+                  onClick = { navigationActions.goBack() },
+                  icon = Icons.AutoMirrored.Filled.ArrowBack,
+                  contentDescription = "GoBack",
+                  modifier = Modifier.testTag("editProfileBackButton"),
+              )
+            })
+      },
```

Comment:
We have a reusable **ScreenTopAppBar** composable in the **MainComponents** file, designed for this type of commonly used top app bar in the app. Consider using it here to reduce code duplication.